introspection-engine/connectors/sql-introspection-connector/src/re_introspection.rs

```diff
@@ -302,10 +302,27 @@ pub fn enrich(old_data_model: &Datamodel, introspection_result: &mut Introspecti
     // potential error: what if there was a db default before and then it got removed, now re-introspection makes it virtual
     // you could not get rid of it
 
+    // restore old model order
+    let mut re_ordered_data_model = Datamodel::new();
+
+    for old_model in old_data_model.models() {
+        if let Some(new_model) = new_data_model.find_model(&old_model.name) {
+            re_ordered_data_model.add_model(new_model.clone());
+        }
+    }
+
+    for new_model in new_data_model.models() {
+        if old_data_model.find_model(&new_model.name).is_none() {
+            re_ordered_data_model.add_model(new_model.clone());
+        }
+    }
+
+    re_ordered_data_model.enums = new_data_model.enums.clone();
+    *new_data_model = re_ordered_data_model;
```

Comment:
I'm a bit concerned that this will drop potential new items (besides models and enums) if/when we add them. I don't know if all the fields in Datamodel are public, so that `Datamodel { models: re_ordered_data_model.models, enums: new_data_model.enums }`.

Alternatively, I think the whole operation could be achieved without cloning and allocation with a `Vec::sort_by`. Something like (pseudocode):

```rust
new_data_models.models.sort_by(|model_a, model_b| {
  let model_a_idx = old_data_model.models.position(|model| model.name == model_a);
  let model_b_idx = ...;

  match (model_a_idx, model_b_idx) {
    (None, None) => Equal,
    (None, Some(_)) => Lesser,
    (Some(_), None) => Greater,
    (Some(a_idx), Some(b_idx)) => a_idx.cmp(b_idx)
  }
})
```

Reply:
Thx for the tips, adjusted it accordingly.